apollo-router/tests/integration_tests.rs

```diff
@@ -572,7 +572,7 @@ async fn query_just_at_recursion_limit() {
 #[tokio::test(flavor = "multi_thread")]
 async fn query_just_under_token_limit() {
     let config = serde_json::json!({
-        "preview_operation_limits": {"parser_max_tokens": 36_usize}
+        "preview_operation_limits": {"parser_max_tokens": 39_usize}
```

Comment:
the query rewriting adds labels adds tokens (a `query` as prefix, and a space at the end)

Reply:
The token count of the rewritten query is an internal detail that should not affect behavior such as which requests are rejected. Assuming the rewrite is never much larger than its input query, I think we should parse rewritten queries without a token limit (and maybe not recursion limit either?)

Reply:
the issue here is that the operation limits apply on the original query that went through the `add_defer_labels` method, because we need to have the labels in both original and filtered query. If we want to fix this, the operation limits should be checked in the `CachingQueryPlanner`, right before adding the labels

Reply:
Max tokens is a parser limit though, not an operation limit checked by an explicit HIR traversal. The first commit of https://github.com/apollographql/router/pull/3356 fixes this.